### PR TITLE
[WIP] first draft to suggest requiring of extensions

### PR DESCRIPTION
**Pull request by maglnet**
As mentioned in #2, it would be nice to see, which dependency should be required to fix an unknown symbol.

This draft checks all loaded extensions for the unknown symbol and suggests requiring an extension if it can be found.

### CODE REVIEW

File: src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        yield from $this->suggestFromLoadedExtensions($symbolName);
+    }
+
+    private function suggestFromLoadedExtensions($symbolName) : \Generator
+    {
+
+        $loadedExtensions = get_loaded_extensions();
+
+        foreach ($loadedExtensions as $extensionName) {
+            $extensionSymbols = (new LocateDefinedSymbolsFromExtensions())->__invoke([$extensionName]);
```
Comment (Ocramius) [inline src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php]: You can create this function outside the loop.

File: src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    public function __invoke($symbolName) : \Generator
+    {
+        yield from $this->suggestFromLoadedExtensions($symbolName);
+    }
+
+    private function suggestFromLoadedExtensions($symbolName) : \Generator
+    {
+
+        $loadedExtensions = get_loaded_extensions();
```
Comment (Ocramius) [inline src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php]: Do this at `__construct`

File: src/ComposerRequireChecker/Cli/CheckCommand.php [hunk 3]
```diff
@@ -90,6 +91,10 @@ protected function execute(InputInterface $input, OutputInterface $output) : int
         $output->writeln("The following unknown symbols were found:");
         foreach ($unknownSymbols as $unknownSymbol) {
             $output->writeln("  " . $unknownSymbol);
+            $suggester = new DependencySuggester();
+            foreach($suggester($unknownSymbol) as $suggestion) {
```
Comment (Ocramius) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: Note: you can create tables for this ;-)

Comment (maglnet) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: I don't get it?!

File: src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php (hunk 2 above)
Comment (maglnet) [inline src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php]: I think I should completely pull this function `suggestFromLoadedExtensions` out of there, and create a standalone `FromLoadedExtensionSuggester`, as I would like to be able to add more "Suggestors" over time (e.g. as mentioned in #7 ).
But for now I'm not sure what would be the best way to accomplish this.

btw. is it "Suggester" or "Suggestor" or is it wrong anyway?! :smile:

File: src/ComposerRequireChecker/Cli/CheckCommand.php (hunk 3 above)
Comment (Ocramius) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: http://symfony.com/doc/current/components/console/helpers/tablehelper.html

File: src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php (hunk 2 above)
Comment (Ocramius) [inline src/ComposerRequireChecker/DependencySuggester/DependencySuggester.php]: maybe "guesser"?

**Pull request merged**